src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -424,6 +429,15 @@ private string GetXrefHref(FilePath file, string uid, int uidCount, bool isRootL
 
                         errors.AddIfNotNull(xrefError);
 
+                        if (xrefSpec != null &&
+                        schema.XrefType != null &&
+                        schema.ContentType == JsonSchemaContentType.Xref &&
+                        (xrefSpec.SchemaType is null ||
+                        !xrefSpec.SchemaType.Equals(schema.XrefType)))
+                        {
+                            errors.AddIfNotNull(Errors.Xref.InvalidXrefType(content, schema.XrefType!, xrefSpec.SchemaType));
```

Comment:
Don't use `!` unless you absolutely need to!
```suggestion
                            errors.Add(Errors.Xref.InvalidXrefType(content, schema.XrefType, xrefSpec.SchemaType));
```

Reply:
OK